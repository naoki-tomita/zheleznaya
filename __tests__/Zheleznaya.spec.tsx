import { Zheleznaya, Component, h } from "../index";

describe("Zheleznaya", () => {
  describe("#createVNode", () => {
    it("should create VNode", () => {
      const target = new Zheleznaya();
      const Inner: Component<{ name: string }> = ({ name }) => {
        return <div>{name}</div>
      }
      const App: Component = () => {
        return (
          <div>
            <Inner name="hoge" />
            <ul>
              <li>top</li>
              {["foo", "bar", "hoge"].map(it => <li>{it}</li>)}
            </ul>
            <div style={{ display: "flex" }} class="my-class">INNNER TEXT</div>
          </div>
        );
      }
      expect(target.createVNode(<App />)).toMatchSnapshot();
    });
  });

  describe("#createHTMLElement", () => {

    it("should create array node with all type node.", () => {
      const target = new Zheleznaya();
      expect(target.createHTMLElement({
        name: "",
        type: "array",
        attributes: {},
        children: [{
          name: "TEXT",
          type: "text",
          attributes: {},
          children: [],
        }, {
          name: "input",
          type: "html",
          attributes: {
            class: "my-class",
            onclick: () => console.log("clicked"),
            style: { display: "flex" },
            value: "INPUT VALUE"
          },
          children: [],
        }, {
          name: "",
          type: "array",
          attributes: {},
          children: [{
            name: "INNER TEXT",
            type: "text",
            attributes: {},
            children: [],
          }],
        }],
      })).toMatchSnapshot()
    });

  });
});
