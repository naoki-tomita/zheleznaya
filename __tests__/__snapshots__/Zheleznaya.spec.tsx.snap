// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zheleznaya #createHTMLElement should create array node with all type node. 1`] = `
Array [
  TEXT,
  <input
    class="my-class"
    style="display: flex;"
  />,
  Array [
    INNER TEXT,
  ],
]
`;

exports[`Zheleznaya #createVNode should create VNode 1`] = `
Object {
  "attributes": Object {},
  "children": Array [
    Object {
      "attributes": Object {},
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [],
          "name": "hoge",
          "type": "text",
        },
      ],
      "name": "div",
      "type": "html",
    },
    Object {
      "attributes": Object {},
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "attributes": Object {},
              "children": Array [],
              "name": "top",
              "type": "text",
            },
          ],
          "name": "li",
          "type": "html",
        },
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "attributes": Object {},
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [],
                  "name": "foo",
                  "type": "text",
                },
              ],
              "name": "li",
              "type": "html",
            },
            Object {
              "attributes": Object {},
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [],
                  "name": "bar",
                  "type": "text",
                },
              ],
              "name": "li",
              "type": "html",
            },
            Object {
              "attributes": Object {},
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [],
                  "name": "hoge",
                  "type": "text",
                },
              ],
              "name": "li",
              "type": "html",
            },
          ],
          "name": "",
          "type": "array",
        },
      ],
      "name": "ul",
      "type": "html",
    },
    Object {
      "attributes": Object {
        "class": "my-class",
        "style": Object {
          "display": "flex",
        },
      },
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [],
          "name": "INNNER TEXT",
          "type": "text",
        },
      ],
      "name": "div",
      "type": "html",
    },
  ],
  "name": "div",
  "type": "html",
}
`;
